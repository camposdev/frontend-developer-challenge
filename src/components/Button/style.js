import styled from "styled-components";

export const ButtonComponent = styled.button`
  display: inline-flex;
  width: ${({ block }) => block ? '100%' : 'auto' };
  align-items: center;
  justify-content: center;
  height: ${({ small }) => small ? "3rem" : "4rem"};
  background-color: white;
  border: 1px solid var(--secondary);
  border-radius: 4px;
  text-decoration: none;
  font-size: 1.4rem;
  white-space: nowrap;
  color: var(--secondary);
  padding: 0 10px;
  box-shadow: ${({ flat }) => flat ? "0 0 0 rgba(0,0,0,0)" : "0 1px 6px rgba(0,0,0,0.16)"};
  cursor: pointer;
  transition: box-shadow 0.3s ease;

  &:hover {
    box-shadow: 0 5px 12px rgba(0,0,0,0.2);
  }
`;