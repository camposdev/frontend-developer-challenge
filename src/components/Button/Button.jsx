import React from "react";
import PropTypes from "prop-types";
import { ButtonComponent } from "./style";

const Button = ({ children, href, ...props }) => {
  if (href) return <ButtonComponent as="a" {...props}>{children}</ButtonComponent>;
  return <ButtonComponent {...props}>{children}</ButtonComponent>;
};

Button.propTypes = {
  block: PropTypes.bool,
  small: PropTypes.bool,
  flat: PropTypes.bool
};

Button.defaultProps = {
  block: false,
  small: false,
  flat: false
}

export default Button;
