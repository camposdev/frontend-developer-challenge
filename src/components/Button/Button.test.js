import React from "react";
import renderer from "react-test-renderer";
import Button from "./Button";

describe("Button test suit", () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<Button>Click me</Button>)
      .toJSON();
    
    expect(tree).toMatchSnapshot();
  });
});