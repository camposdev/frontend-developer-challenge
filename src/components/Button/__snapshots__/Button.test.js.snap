// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button test suit renders correctly 1`] = `
<button
  className="sc-bdVaJa kmThUw"
>
  Click me
</button>
`;
